javascript/features/killtime/killtime_manager.js

```diff
@@ -80,13 +123,31 @@ class KilltimeManager {
         if (player != null)
             stopMessage = Message.KILLTIME_ADMIN_STOPPED;
 
+        this.removePlayerKillTimeWeapons();
         this.announce_().announceToPlayers(stopMessage);
         this.announceWinner_();
 
         this.scoreMap_.clear();
 
     }
 
+    //Gives all currently online players the weapon
+    removePlayerKillTimeWeapons() {        
+        if(this.weapon_ === null || this.weapon_ === undefined)
+            return;
+
+        server.playerManager.forEach(player => {
+            if (player.isNonPlayerCharacter())
+                return;
+            if(player.activity !== Player.PLAYER_ACTIVITY_NONE)
+                return;
+
+            player.removeWeapon(this.weapon_);            
```

Comment:
What if the player has just bought the weapon already? Imo we should give the player the weapon like we now do and set it back at the end to the amount of ammo we had. If you really want to make it complicated you could also subtract the amount of shot bullets...

Reply:
There's probably more to it than just checking if the weapon is currently owned. 

However I ditched this too soon and will have a look into it.

Reply:
This is more effort than it's worth right now.
We'd need a weapon -> weapon slot mapping for this as you can only get weapon for a specific weapon slot. 

I removed the spawnweapons instead to avoid this. I expected admins not to use them anyway but it's better to also remvoe it from the lest then.